packages/SwingSet/docs/vat-environment.md

```diff
@@ -0,0 +1,120 @@
+# The Vat Runtime JS Environment
+
+SwingSet JavaScript Vats are containers that run code in a confined and resource-limited environment, with orthogonal persistence and eventual-send-based access to external resources.
+
+The specific flavor of JavaScript provided by a vat is a frozen SES `Compartment`, with a few additions. Most JS environments (Node.js, web browsers) provide a combination of the baseline JavaScript [language globals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects) (`Object`, `Array`, etc) plus some number of "host objects". [Web browsers](https://developer.mozilla.org/en-US/docs/Web/API) offer things like `window`, `fetch`, `WebAssembly`, and `localStorage`, while [Node.js includes](https://nodejs.org/dist/latest-v14.x/docs/api/globals.html) ones like `require`, `process`. and `Buffer`. In general, all IO is provided by the host objects.
+
+SES is a safe deterministic subset of "strict mode" JavaScript, which means it does not include any of these IO objects (they would provide ambient authority, which is not "safe"). It also modifies a few built-in objects to remove non-determinism.
+
+The full list of SES globals (as of SES-0.8.0) is defined in [the SES source code](https://github.com/Agoric/SES-shim/blob/SES-v0.8.0/packages/ses/src/whitelist.js). It includes things like `Object`, `Array`, `Number`, `Map` and `WeakMap`, `Number` and `BigInt`. This is a subset of the baseline JavaScript language globals.
+
```

Comment:
(@dckc): this list is pretty much a click here link. I suggest using full list of SES globals as the link text.
(@warner): hm, I don't know how to make that scan correctly, but how about `The full list of SES globals is defined in (link text: the SES source code)` ?
(@tyg): How about "Click here for the (link text: full list of SES Globals, as of SES-0.8.0)"
2) "It includes Object, ..., BigInt and others."
3) Is "language" needed in "JavaScript language globals"? They all should know JS is a language...

Hm, I think @dckc would similarly object to a sentence that literally instructs the reader to "click here". Since the rest of the paragraph is about the contents/definition of that list, I think I'll stick to using `is defined` as the lingering active verb of the introductory sentence.

2 sounds good. For 3, I meant to suggest something about the official specification. I should make that more explicit. How about:

The full list of SES globals (as of SES-0.8.0) is defined in [the SES source code](https://github.com/Agoric/SES-shim/blob/SES-v0.8.0/packages/ses/src/whitelist.js). It includes the core language objects: `Object`, `Array`, `Number`, `Map` and `WeakMap`, `Number`, `BigInt`, and others. This is a subset of the globals defined by the baseline JavaScript language specification.

Reply:
It might be good to make the link permanent (press `y` while on that page). I’d like to have my hands free to move that file around.

Reply:
Looks good to me.

Reply:
> It might be good to make the link permanent (press `y` while on that page). I’d like to have my hands free to move that file around.

The link includes the `SES-v0.8.0` tag.. isn't that sufficient? I'll make sure all the other links either include a tag or a full commit hash.

Reply:
link text: the SES source code works for me.